ray-curator/ray_curator/stages/video/filtering/motion_filter.py

```diff
@@ -0,0 +1,164 @@
+
+    @property
+    def resources(self) -> Resources:
+        return Resources(gpu_memory_gb=self.gpu_memory_gb)
+
+    def process(self, task: VideoTask) -> VideoTask:
+        video: Video = task.data
+
+
+        passing_clips = []
+        for clip in video.clips:
+            if not clip.decoded_motion_data:
+                if self.verbose:
+                    logger.warning(f"Clip {clip.uuid} has no decoded motion data.")
+                fake_score = -1.0
+                motion_info = motion_backend.MotionInfo(
+                    fake_score < self.global_mean_threshold or fake_score < self.per_patch_min_256_threshold,
+                    fake_score,
+                    fake_score,
+                )
+            else:
+                motion_info = motion_backend.check_if_small_motion(
+                    clip.decoded_motion_data.frames,
+                    clip.decoded_motion_data.frame_size,
+                    global_mean_threshold=self.global_mean_threshold,
+                    per_patch_min_256_threshold=self.per_patch_min_256_threshold,
+                    use_gpu=self.gpu_memory_gb > 0,
+                    batch_size=self.motion_filter_batch_size,
+                )
+
+            clip.decoded_motion_data = None
+            clip.motion_score_global_mean = motion_info.global_mean
+            clip.motion_score_per_patch_min_256 = motion_info.per_patch_min_256
+            if motion_info.is_small_motion:
+                if self.score_only:
+                    passing_clips.append(clip)
+                else:
+                    video.filtered_clips.append(clip)
+                    video.clip_stats.num_filtered_by_motion += 1
+                if self.verbose:
+                    logger.info(
+                        f"Clip {clip.uuid} has motion score global mean {clip.motion_score_global_mean:.5f}"
+                        f"<{self.global_mean_threshold} or per-patch min 256 "
+                        f"{clip.motion_score_per_patch_min_256:.6f}<{self.per_patch_min_256_threshold}, "
+                        f"skipped.",
+                    )
+            else:
+                passing_clips.append(clip)
+                if self.verbose:
+                    logger.info(
+                        f"Clip {clip.uuid} has motion score global mean {clip.motion_score_global_mean:.5f}"
+                        f">={self.global_mean_threshold} and per-patch min 256 "
+                        f"{clip.motion_score_per_patch_min_256:.6f}>={self.per_patch_min_256_threshold}, "
+                        f"kept.",
+                    )
+        video.clips = passing_clips
+
+        logger.info(
+            f"Video {video.input_video} chunk-{video.clip_chunk_index} has "
+            f"{len(video.clips)}/{len(video.filtered_clips) + len(video.clips)} clips "
+            "passed/filtered",
+        )
+
+        # TODO: free memory periodically
```

Comment:
Can we add your name to the todo so that it;s clear? Also why can't we just put the empty cache ?

Reply:
We could invoke `torch.cuda.empty_cache` after each `process` call , but it would be too excessive. I think we need some mechanism from a more base level that call this less frequent .